api/authentication.go

```diff
@@ -299,6 +300,14 @@ func checkAuth(w http.ResponseWriter, r *http.Request, authRequired bool) (token
 		return nil, false
 	}
 
+	// Abort if the token is expired.
+	if token.ValidUntil != nil && token.ValidUntil.After(time.Now()) {
```

Comment:
I think you meant `time.Now().After(token.ValidUntil)`.

Reply:
This might better belong into `checkAPIKey()` - I'm not even sure it would be triggered here.